brdnet/run_plier.R

```diff
@@ -10,13 +10,32 @@ library('biomaRt')
 # Install PLIER if it isn't already present
 if (!require('PLIER',character.only = TRUE)) {
 	library('devtools')
-	install_github('wgmao/PLIER')
+	tryCatch({
+		install_github('wgmao/PLIER')
+	},
+	error=function(e){
+		# For some reason devtools struggles to find the tar installation, this should fix it
+		Sys.setenv(TAR = "/bin/tar")
+		install_github('wgmao/PLIER')
+	})
+
 }
 library('PLIER')
 
-
 data.dir <- '../data'
 
+args <- commandArgs(trailingOnly=TRUE)
+
+if(length(args) < 2){
+	print('As no value of k was passed in, it will be estimated from the data')
```

Comment:
For my own knowledge, I'm curious if these were checks that existed in the PLIER scripts or are these ones that you generated?  If you generated them, how did you come up with these?

Reply:
The PLIER vignettes calculate the value of `k` using the function `PLIER::num.pc`. Since I was testing whether I would get better performance by increasing or decreasing the number of PCs, having an optional `k` argument in `run_plier.R` made it possible to create embedding matrices for several values of `k` in parallel